pkg/terraform/terraform.go

```diff
@@ -80,23 +80,20 @@ func (m *Mixin) getOutput(outputName string) ([]byte, error) {
 	if err != nil {
 		return []byte{}, err
 	}
+	// If the output is a string then don't re-encode it, just return the decoded
+	// json string. Re-encoding the string will wrap it in quotes which breaks
+	// the compabiltity with -raw
+	if outStr, ok := outDat.(string); ok {
+		return []byte(outStr), nil
+	}
 	buffer := &bytes.Buffer{}
 	encoder := json.NewEncoder(buffer)
 	encoder.SetEscapeHTML(false)
 	err = encoder.Encode(outDat)
 	if err != nil {
 		return []byte{}, err
 	}
-	outJSON := bytes.TrimRight(buffer.Bytes(), "\n")
-	// If the output data is of type string then do one last unmarshal to json
-	// string so that the outer quotes are stripped, otherwise return the raw
-	// json syntax directly.
-	var outString string
-	if err = json.Unmarshal(outJSON, &outString); err == nil {
-		return []byte(outString), nil
-	}
-	// For all other JSON types we'll return the raw JSON syntax directly.
-	return outJSON, nil
+	return bytes.TrimRight(buffer.Bytes(), "\n"), nil
```

Comment:
Does this mean we don't need to trim the new line character when the output is a string?

Reply:
No since we're not re-encoding it. The json.Encode is what puts the newline on the end:
From the docs - Encode writes the JSON encoding of v to the stream, followed by a newline character.

Reply:
I did some testing. It looks like `json.Unmarshal()` removes the new line character. We should be good. Thanks for answering my questions :)

Reply:
Yep that's what I was in the middle of validating! Beat me to it :)